Adopt smart pointers in WebInspectorUI.cpp, WebProcessCache.cpp, WebPageProxy.cpp, WebPageInspectorController.cpp

jelee53 opened the pull request:
#### 233dd7041db960186ad176798d4878f8389d4288
<pre>
Adopt smart pointers in WebInspectorUI.cpp, WebProcessCache.cpp, WebPageProxy.cpp, WebPageInspectorController.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=295307">https://bugs.webkit.org/show_bug.cgi?id=295307</a>
<a href="https://rdar.apple.com/154808457">rdar://154808457</a>

Reviewed by Per Arne Vollan.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebKit/Shared/WebEvent.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
(WebKit::WebPageInspectorController::destroyInspectorTarget):
(WebKit::WebPageInspectorController::sendMessageToInspectorFrontend):
(WebKit::WebPageInspectorController::didCommitProvisionalPage):
(WebKit::WebPageInspectorController::addTarget):
(WebKit::WebPageInspectorController::browserExtensionsEnabled):
(WebKit::WebPageInspectorController::browserExtensionsDisabled):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.h:
(WebKit::WebPageInspectorController::checkedTargetAgent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::processNextQueuedMouseEvent):
(WebKit::WebPageProxy::mouseEventHandlingCompleted):
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::addProcessIfPossible):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::logDiagnosticEvent):

Canonical link: <a href="https://commits.webkit.org/297007@main">https://commits.webkit.org/297007@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f32dc6c236c6656e92c5f0f1cfdf1712fe19aa9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29903 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20335 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116268 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/60492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30582 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38490 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/83819 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/60492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113192 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24394 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99274 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64268 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/23754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60062 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93768 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17468 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119058 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37284 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/27641 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/92794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37656 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95544 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92620 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37611 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15341 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/33173 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17786 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37178 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42649 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36840 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40180 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38549 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
```diff
@@ -175,13 +175,13 @@ void WebPageInspectorController::destroyInspectorTarget(const String& targetId)
     if (it == m_targets.end())
         return;
-    m_targetAgent->targetDestroyed(*it->value);
+    CheckedPtr { m_targetAgent }->targetDestroyed(*it->value);
```
I wonder if we can make m_targetAgent a `const UniqueRef` instead and use `lazyInitialize` given the feedback I got from @charliewolfe elsewhere.

pvollan commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
```diff
@@ -175,13 +175,13 @@ void WebPageInspectorController::destroyInspectorTarget(const String& targetId)
 void WebPageInspectorController::sendMessageToInspectorFrontend(const String& targetId, const String& message)
 {
-    m_targetAgent->sendMessageFromTargetToFrontend(targetId, message);
+    CheckedPtr { m_targetAgent }->sendMessageFromTargetToFrontend(targetId, message);
```
Should we create a helper function for this?

pvollan commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
```diff
@@ -216,15 +216,16 @@ void WebPageInspectorController::didCommitProvisionalPage(WebCore::PageIdentifie
     ASSERT(newTarget);
     newTarget->didCommitProvisionalTarget();
-    m_targetAgent->didCommitProvisionalTarget(oldID, newID);
+    targetAgent->didCommitProvisionalTarget(oldID, newID);
```
Should we use `checkedTargetAgent()` here too?

pvollan commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
```diff
@@ -216,15 +216,16 @@ void WebPageInspectorController::didCommitProvisionalPage(WebCore::PageIdentifie
     // FIXME: <https://webkit.org/b/202937> do not destroy targets that belong to the committed page.
     for (auto& target : m_targets.values())
-        m_targetAgent->targetDestroyed(*target);
+        targetAgent->targetDestroyed(*target);
```
Ditto.

pvollan reviewed (approved):
R=me.

jelee53 committed:
Adopt smart pointers in WebInspectorUI.cpp, WebProcessCache.cpp, WebPageProxy.cpp, WebPageInspectorController.cpp
https://bugs.webkit.org/show_bug.cgi?id=295307
rdar://154808457

Reviewed by Per Arne Vollan.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebKit/Shared/WebEvent.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
(WebKit::WebPageInspectorController::destroyInspectorTarget):
(WebKit::WebPageInspectorController::sendMessageToInspectorFrontend):
(WebKit::WebPageInspectorController::didCommitProvisionalPage):
(WebKit::WebPageInspectorController::addTarget):
(WebKit::WebPageInspectorController::browserExtensionsEnabled):
(WebKit::WebPageInspectorController::browserExtensionsDisabled):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.h:
(WebKit::WebPageInspectorController::checkedTargetAgent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::processNextQueuedMouseEvent):
(WebKit::WebPageProxy::mouseEventHandlingCompleted):
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::addProcessIfPossible):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::logDiagnosticEvent):

Canonical link: https://commits.webkit.org/297007@main
